packages/@sfx/dom-events-plugin/test/unit/browser/events-browser-plugin.test.ts

```diff
@@ -0,0 +1,110 @@
+import { expect, sinon, spy, stub } from '../../utils';
+import DomEventsPlugin from '../../../src/dom-events-plugin';
+
+describe('domEventsPlugin', () => {
+  let domEventsPlugin: any;
+
+  beforeEach(() => {
+    domEventsPlugin = new DomEventsPlugin();
+  });
+
+  describe('constructor()', () => {
+    it('should create a DomEventsPlugin instance with default options', () => {
+      const defaultOptions = { window: window, CustomEvent: CustomEvent };
+
+      expect(domEventsPlugin.options).to.deep.equal(defaultOptions);
+    });
+
+    it('should throw an error if an invalid window option is provided', () => {
+      const invalidOptions = {
+        window: undefined,
+        CustomEvent: CustomEvent,
+      };
+
+      expect(() => new DomEventsPlugin(invalidOptions)).to.throw();
+    });
+
+    it('should throw an error if an invalid CustomEvent constructor is provided', () => {
+      const invalidOptions = {
+        windowL: Window,
+        CustomEvent: undefined,
+      };
+
+      expect(() => new DomEventsPlugin(invalidOptions)).to.throw();
+    });
+
+    it('should combine default options and provided options', () => {
+      const defaultOptions = { window: window, CustomEvent: CustomEvent };
+      const options: any = { a: 'b', c: 'd', };
+
+      domEventsPlugin = new DomEventsPlugin(options);
+
+      expect(domEventsPlugin.options).to.deep.equal({ ...defaultOptions, ...options });
+    });
+
+    it('should override default options', () => {
+      const options: any = {
+        foo: 'bar',
+        baz: 'qux',
+        window: {
+          windowProp: 'windowPropValue',
+        },
+      };
+      domEventsPlugin = new DomEventsPlugin(options);
+
+      expect(domEventsPlugin.window).to.deep.equal(options.window);
```

Comment:
I don't think you want `deep` here. You should also be checking `domEventsPlugin.options.window` instead of `domEventsPlugin.window`. The test for the latter should be another test.

Reply:
Updated.